app/src/main/java/com/example/tasky/feature_agenda/presentation/Agenda.kt

```diff
@@ -260,10 +270,15 @@ fun AgendaContent(
                     verticalArrangement = Arrangement.spacedBy(8.dp)
                 ) {
                     if (!agendaItems.isNullOrEmpty()) {
-                        items(
-                            items = agendaItems,
-                            key = { it.id }
-                        ) { agendaItem ->
+                        itemsIndexed(
+                            items = agendaItems
+                        ) { index, agendaItem ->
+                            if (index > 0 && agendaItem.startTime.toLocalDate() == needlePosition.toLocalDate() && agendaItem.startTime.isAfter(
+                                    needlePosition
+                                ) && agendaItems[index - 1].startTime.isBefore(needlePosition)
+                            ) {
+                                Needle()
```

Comment:
I personally created a separate class that treats the Needle as its own element in the list:
```
sealed interface AgendaItemUi {
    data class Item(val item: AgendaItem) : AgendaItemUi
    data object Needle : AgendaItemUi
}
```
Then the UI will get a List<AgendaItemUi> and the ViewModel can handle the logic to decide where the needle is. This has the advantage that you can easily animate item changes, e.g. when time passes and the needle moves past the next item.